hi-ml-multimodal/src/health_multimodal/image/model/resnet.py

```diff
@@ -20,7 +22,15 @@ class ResNetHIML(ResNet):
     def __init__(self, **kwargs: Any) -> None:
         super().__init__(**kwargs)
 
-    def forward(self, x: torch.Tensor) -> torch.Tensor:
+    def forward(self, x: torch.Tensor,
+                return_skip: bool = False) -> Union[torch.Tensor, TypeSkipConnections]:
```

Comment:
As far as I know ResNet doesn't have skip connections as in U-Net. These are intermediate layers. Maybe you could rename it as `return_intermediate_layers`

Reply:
Another option could be `return_all_feature_maps`, as in `pl_bolts`: https://github.com/Lightning-AI/lightning-bolts/blob/64c4bac1aabeea88a4d577e2522752f5ec685e9a/pl_bolts/models/self_supervised/resnets.py#L261

I suspect their implementation might be more efficient in terms of memory. If we don't care about intermediate activations, maybe overwriting `x0` is better than storing the values of `x0`, `x1`, etc. in memory. I guess that's why a lot of code goes

```
x = layer1(x)
x = layer2(x)
etc
```

Reply:
They are all stored in memory as part of the computation graph for backprop unless the `no_grad` decorator is passed as in inference mode only. I guess memory wouldn't be a big concern at test time.

We didn't use pl_bolts on purpose, their 3party dependencies are not updated on a regular basis.

Reply:
Thanks Fernando. I think either `return_all_feature_maps` or `return_intermediate_layers` should work fine. Let's update it and then we can merge the PR in soon.